tests/lib/Controller/SettingsController.php

```diff
@@ -596,4 +596,29 @@ public function withCorsAnnotation(): DataResponse {
 	public function withCorsAttribute(): DataResponse {
 		return new DataResponse();
 	}
+
+	/**
+	 * A deprecated route
+	 *
+	 * @param bool|true $yesOrNo boolean or true  This parameter will be removed in a future version
+	 * @return DataResponse<Http::STATUS_OK, list<empty>, array{}>
+	 * @deprecated 1.0.0 Use {@see SettingsController::trueFalseParameter()} instead
+	 *
+	 * 200: Admin settings updated
+	 */
+	public function deprecatedRoute(bool $yesOrNo): DataResponse {
+		return new DataResponse();
+	}
+
+	/**
+	 * A deprecated parameter does not deprecate the method
+	 *
+	 * @param bool|true $yesOrNo boolean or true @deprecated This parameter will be removed in a future version
+	 * @return DataResponse<Http::STATUS_OK, list<empty>, array{}>
+	 *
+	 * 200: Admin settings updated
+	 */
+	public function deprecatedParameter(bool $yesOrNo): DataResponse {
```

Comment:
A combination of deprecated route with a deprecated and non deprecated parameter should be there as well.